File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int size = brown + yellow;
        List<Integer> candidates = new ArrayList<>();

        for(int i=1; i<=size; i++) {
            if(size%i == 0) candidates.add(i);
        }

        int answer1 = 1;
        for(Integer i : candidates) {
            if((i-2)*((size/i)-2)==yellow) {
                answer1 = i;
            }
        }
        return new int[]{answer1, size/answer1};
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.go

func solution(brown, yellow int) []int {
	//전체 타일 수
	all := brown + yellow

	answer := make([]int, 0, 2)
	//약수만 진행
	for i := all; i >= 1; i-- {
		if all%i != 0 {
			continue
		}

		w := i
		l := all / i

		if brown == (2*w + 2*l - 4) {
			answer = append(answer, w)
			answer = append(answer, l)
			break
		}

	}
	return answer
}